src/qibocal/tree/operation.py

```diff
@@ -0,0 +1,19 @@
+from enum import Enum
+
+
+def command_1():
+    print("command 1")
+
+
+def command_2():
+    print("command 2")
+
+
+def command_3():
+    print("command 3")
+
+
+class Operation(Enum):
+    command_1 = command_1  # TODO: this is not working
+    command_2 = command_2
+    command_3 = command_3
```

Comment:
```suggestion
from dataclasses import dataclass
from typing import Callable

@dataclass
class Routine:
    name: str
    routine: Callable

def _command_1():
    print("command 1")

command_1 = Routine("1", routine=_command_1)

def _command_2():
    print("command 2")

command_2 = Routine("2", routine=_command_2)

def _command_3():
    print("command 3")

command_3 = Routine("3", routine=_command_3)

class Operation(Enum):
    command_1 = command_1
    command_2 = command_2
    command_3 = command_3
```

Let's start like this, test if it's working :)

Reply:
In principle, we could use one element tuples. But let's use immediately something that looks closer to what I have in mind for the final product.

Reply:
`Operation.command_1.routine()` should invoke your function.

If you have a string for the name of the element (the LHS inside the class, not the attribute) you can use it as `Operation['command_1']`, obtaining the same object as `Operation.command_1`.
However, whenever you can avoid the string.